### PR TITLE
make Azure File CSI migration as GA in 1.26

### PR DESCRIPTION
**Pull request by andyzhangx**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
make Azure File CSI migration as GA in 1.26

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
/assign @msau42 @xing-yang 
/priority important-soon
/sig cloud-provider
/area provider/azure
/triage accepted

#### Does this PR introduce a user-facing change?

```release-note
make Azure File CSI migration as GA in 1.26
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
make Azure File CSI migration as GA in 1.26
```

### CODE REVIEW

**Comment by bertinatto** [comment]
@andyzhangx @gnufied @jsafrane @xing-yang 
Should we hold this until we promote [FSGroup](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/2317-fsgroup-on-mount) to GA?

**Comment by bertinatto** [comment]
@andyzhangx, the release note in the PR description seems incorrect, it should be GA in 1.26.

**Comment by andyzhangx** [comment]
> @andyzhangx @gnufied @jsafrane @xing-yang Should we hold this until we promote [FSGroup](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/2317-fsgroup-on-mount) to GA?

@bertinatto per https://github.com/kubernetes/enhancements/pull/3550, fsgroup mount should be also GA in 1.26, so it's good to go?

**Comment by andyzhangx** [comment]
cc @bridgetkromhout

**Comment by bertinatto** [comment]
> > @andyzhangx @gnufied @jsafrane @xing-yang Should we hold this until we promote [FSGroup](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/2317-fsgroup-on-mount) to GA?
> 
> @bertinatto per [kubernetes/enhancements#3550](https://github.com/kubernetes/enhancements/pull/3550), fsgroup mount should be also GA in 1.26, so it's good to go?

That's correct, FSGroup mount will be GA in 1.26, however, the code changes haven't been made yet.

**Comment by andyzhangx** [comment]
> @gnufied

@gnufied are you going to make fsgroup mount GA change? thanks.

**Comment by gnufied** [comment]
@bertinatto is working on it - https://github.com/kubernetes/kubernetes/pull/113225

File: pkg/features/kube_features.go [hunk 1]
```diff
@@ -824,7 +825,7 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 
 	CSIMigrationAzureDisk: {Default: true, PreRelease: featuregate.GA, LockToDefault: true}, // On by default in 1.23 (requires Azure Disk CSI driver)
 
-	CSIMigrationAzureFile: {Default: true, PreRelease: featuregate.Beta}, // On by default in 1.24 (requires Azure File CSI driver)
+	CSIMigrationAzureFile: {Default: true, PreRelease: featuregate.GA}, // On by default in 1.24 (requires Azure File CSI driver)
```
Comment (jsafrane) [inline pkg/features/kube_features.go:828]: Please add comment `remove in 1.28`

**Comment by jsafrane** [comment]
FSGroup mount is GA!
/approve

File: pkg/features/kube_features.go (hunk 1 above)
Comment (jsafrane) [reply pkg/features/kube_features.go:828]: Or better, replace the existing comment with `// remove in 1.28`

**Comment by dims** [comment]
@andyzhangx looks like one suggestion in there about a comment when to remove the flag. can you please do that?

**Comment by jsafrane** [comment]
Andy is on PTO, I'll fix the comment in a subsequent PR
/lgtm

**Pull request merged**